chemprop/train/make_predictions.py

```diff
@@ -34,6 +34,12 @@ def make_predictions(args: PredictArgs, smiles: List[List[str]] = None) -> List[
         raise ValueError('Features were used during training so they must be specified again during prediction '
                          'using the same type of features as before (with either --features_generator or '
                          '--features_path and using --no_features_scaling if applicable).')
+    
+    # Same number of molecules must be used in training as in making predictions
+    assert train_args.number_of_molecules == args.number_of_molecules, (
```

Comment:
It would be better to raise an exception here (perhaps `ValueError`). Assertions should only be used to ensure situations which should always be true.

Reply:
This is good style/function input, thanks